lib/semant.ml

```diff
@@ -162,7 +174,8 @@ let check (binds, funcs, stmts): sprogram =
   
   in
   let check_func fn = 
-    if Hashtbl.mem all_funcs fn.fname || Hashtbl.mem (List.hd !all_scopes) fn.fname then raise (Failure duplicate_func_err)
+    if fn.fname = "say" then raise (Failure reserved_function_say_err)
```

Comment:
We could check if the user defined their own `say` function in the same place where we check for `main`? So like `fun fn -> fn.fname = "main" || fn.fname = "say"`?

Reply:
This is a good idea